src/clr-angular/forms/input/input-container.ts

```diff
@@ -23,19 +28,42 @@ import { NgControlService } from '../common/providers/ng-control.service';
             <ng-content select="clr-control-error" *ngIf="invalid"></ng-content>
         </div>
     `,
-  host: { '[class.clr-form-control]': 'true' },
-  providers: [IfErrorService, NgControlService],
+  host: {
+    '[class.clr-form-control]': 'true',
+    '[class.clr-row]': '!isVertical()',
+  },
+  providers: [IfErrorService, NgControlService, ControlIdService],
 })
-export class ClrInputContainer implements OnDestroy {
+export class ClrInputContainer implements DynamicWrapper, OnDestroy {
   subscription: Subscription;
   invalid = false;
+  _dynamic = false;
+  @ContentChild(ClrLabel) label: ClrLabel;
 
-  constructor(private ifErrorService: IfErrorService) {
+  constructor(private ifErrorService: IfErrorService, @Optional() private layoutService: LayoutService) {
     this.subscription = this.ifErrorService.statusChanges.subscribe(control => {
       this.invalid = control.invalid;
     });
   }
 
+  controlClass() {
+    const controlClasses = [];
+    if (this.invalid) {
+      controlClasses.push('clr-error');
+    }
+    if (!this.isVertical()) {
+      controlClasses.push('clr-col-md-10 clr-col-xs-12');
```

Comment:
Same question here. What happens if the user wants to override the classes?

Reply:
I'll add a check to only add if there are no existing col- classes